starport_template/lib/pages/wallet_details_page.dart

```diff
@@ -1,14 +1,110 @@
+import 'package:cosmos_ui_components/components/content_state_switcher.dart';
+import 'package:cosmos_ui_components/components/template/cosmos_balance_card.dart';
+import 'package:cosmos_ui_components/components/template/cosmos_balance_heading.dart';
+import 'package:cosmos_ui_components/components/template/cosmos_wallets_list_view.dart';
 import 'package:flutter/material.dart';
+import 'package:mobx/mobx.dart';
+import 'package:starport_template/entities/balance.dart';
+import 'package:starport_template/entities/denom.dart';
+import 'package:starport_template/starport_app.dart';
+import 'package:starport_template/widgets/send_money_sheet.dart';
 
-// TODO
-class WalletDetailsPage extends StatelessWidget {
-  const WalletDetailsPage({Key? key}) : super(key: key);
+class WalletDetailsPage extends StatefulWidget {
+  final WalletInfo walletInfo;
+
+  const WalletDetailsPage({Key? key, required this.walletInfo}) : super(key: key);
+
+  @override
+  _WalletDetailsPageState createState() => _WalletDetailsPageState();
+}
+
+class _WalletDetailsPageState extends State<WalletDetailsPage> {
+  Observable<List<Balance>>? get balancesList => StarportApp.walletsStore.balancesList;
+
+  Observable<bool> get isBalancesLoading => StarportApp.walletsStore.isBalancesLoading;
+
+  Observable<bool> get isSendMoneyLoading => StarportApp.walletsStore.isSendMoneyLoading;
+
+  Observable<bool> get isError => StarportApp.walletsStore.isError;
+
+  @override
+  void initState() {
+    super.initState();
+    _fetchWalletBalances();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-      body: Center(
-        child: Text("Wallet Details Page"),
+    return Scaffold(
+      body: SafeArea(
+        child: Center(
+          child: ContentStateSwitcher(
+            contentChild: Column(
+              children: [
+                ListTile(
+                  title: const Text('Wallet address'),
+                  subtitle: Text(widget.walletInfo.address),
+                ),
+                const Divider(),
+                const Padding(padding: EdgeInsets.only(top: 16)),
+                BalanceHeading(),
+                if (balancesList != null)
+                  Padding(
+                    padding: const EdgeInsets.all(8.0),
+                    child: Column(
+                      children: balancesList!.value
+                          .map(
+                            (balance) => BalanceCard(
+                              denomText: balance.denom.text,
+                              amountDisplayText: balance.amount.value.toString(),
+                              onTransferPressed: () {
+                                final denom = Denom(balance.denom.text);
+                                openSendMoneySheet(denom);
+                              },
```

Comment:
lets extract the logic to separate method
```suggestion
                              onTransferPressed: _transferPressed,
```

Reply:
✅